src/axom/core/FlatMap.hpp

```diff
@@ -437,6 +437,24 @@ class FlatMap : detail::flat_map::SequentialLookupPolicy<typename Hash::result_t
   template <typename InputIt>
   void insert(InputIt first, InputIt last);
 
+  /*!
+   * \brief Inserts a range of key-value pairs into the FlatMap.
+   *
+   *  If the key already exists in the FlatMap, insertion is skipped.
+   *  Otherwise, the key-value mapping is inserted into the FlatMap.
+   *
+   * \param [in] first the beginning of the range of pairs
+   * \param [in] last the end of the range of pairs
+   *
+   * \tparam ExecSpace the execution space in which to perform the batched
+   *                   construction
+   *
+   * \pre InputIt is a random-access iterator
+   * \pre allocator is accessible from ExecSpace
+   */
+  template <typename ExecSpace, typename InputIt>
+  void insert(InputIt first, InputIt last);
```

Comment:
Should this method and possible others return a `bool`, for example, indicating whether the insertion was successful? Users may want to use that for debugging, checking behavior of their code.

Reply:
I was thinking something like: return false if insertion is skipped, else return true.

Reply:
For any or all of the range of elements being inserted?

Reply:
Yeah, I should probably clarify that. The behavior is if a key-value pair in the inserting range shares a key with a pair already in the `FlatMap`, we just overwrite the value. For the standard single-KV pair `insert` methods, `true` means we inserted a new pair as opposed to just overwriting the value of an existing pair.

While `std::unordered_map`'s range insert just specifies a `void` return, I agree that we should return something.  How does returning an `int` count of newly-inserted (as opposed to overwritten/duplicated) key-value pairs sound?

Reply:
@publixsubfan your suggestion of returning int seems like a good idea along with accompanying expanded description like you posted here.  Thank you for the clarification.

Reply:
I'd suggest keeping the return type as `void`.

I checked the `std::unordered_map` and `std::flat_map` (C++23) and both have `insert(InputIt first, InputIt last) -> void`
* https://en.cppreference.com/w/cpp/container/unordered_map/insert.html
* https://en.cppreference.com/w/cpp/container/flat_map/insert.html

Reply:
`std::unordered_map::insert` method for single element insertion does return a `std::pair` with a boolean component indicating whether a new element was inserted, which can be considered inconsistent IMO.

I suggest beefing up the method comment to indicate that behavior follows `std::unordered_map` semantics.

Reply:
I've just updated the comments for now.